### ISSUE TITLE
HaxePunk 4.0 Release

### ISSUE THREAD

**Issue by bendmorris** [HaxePunk/HaxePunk#542]
A lot of work has been done on the dev branch, and we need to finalize what needs to be done before release, and what can be postponed, so we can finally get out this (very late) release.

I'll leave some pending tasks as comments to this issue. Add a :+1: or :-1: to each if you have an opinion on prioritizing these features, or comment to add your own ideas.

**Comment by bendmorris** [comment]
Set up an official blog

**Comment by bendmorris** [comment]
Update/migrate to a new forum (Discourse is one option)

**Comment by bendmorris** [comment]
More advanced asset caching solution for loading/unloading assets

**Comment by bendmorris** [comment]
Improve Emitter and add support for the .pex format

**Comment by bendmorris** [comment]
UTF8 text support for BitmapText

**Comment by bendmorris** [comment]
Better/automated Gamepad button handling

**Comment by bendmorris** [comment]
Add origins to all graphic classes

**Comment by bendmorris** [comment]
Add a Kha backend as an alternative to OpenFL/NME

**Comment by IkonOne** [comment]
Overhaul collision to provide detailed collision information and some form of spatial partitioning.

**Comment by IkonOne** [comment]
Update signals to use a [PriorityQueue](https://github.com/HaxePunk/ai/blob/master/com/haxepunk/ds/PriorityQueue.hx) instead of arrays in the background.

**Comment by IkonOne** [comment]
Animation system similar to [Flambe](https://github.com/aduros/flambe/tree/master/src/flambe/script).

**Comment by IkonOne** [comment]
Revisit namespace organization.  (eg. Is Signal not a util? Is Position math?  Cursor graphics?)

**Comment by FlamingQuasar** [comment]
Solving of atlas exceptions problems with hardware Renderer (openfl.gl.GL)
while compiling game on windows (on no graphics-card laptop)

2018-01-25 8:16 GMT+04:00 Erin M Gunn <notifications@github.com>:

**Comment by IkonOne** [comment]
@FlamingQuasar could you open a [new issue](https://github.com/HaxePunk/HaxePunk/issues/new) for this one and provide more details such as the specific error you are receiving and what the code is doing when the error occurs.  A code example would be great if you can put one together.

**Comment by SunShiranui** [comment]
Multiple cameras (for split-screen and similar uses)

**Comment by cjmxp** [comment]
It is recommended to keep openfl after adding Kha. There are some problems in KHA, such as localization language, text input IME, and so on.

**Comment by cjmxp** [comment]
Is it possible to add Multi rendering target

**Comment by eliolanda** [comment]
Add 'delay' property for Tween Class

**Comment by cjmxp** [comment]
Add RenderTexture support

**Comment by cjmxp** [comment]
Add 2d Light.
